zebrad/src/components/mempool.rs

```diff
@@ -626,3 +829,9 @@ impl Service<Request> for Mempool {
         }
     }
 }
+
+impl Drop for Mempool {
+    fn drop(&mut self) {
+        self.disable_metrics();
```

Comment:
At the moment, if the `progress-bar` feature is off, disabling metrics does nothing. But in future, other kinds of metrics might need to be disabled, so I don't think it should be behind a feature.

(And it might cause confusing bugs if it was.)